Docs/llm document structure

davidberenstein1957 opened the pull request:
# Description

A new structure to facilitate LLM docs.

**Type of change**


- [X] Documentation update

**How Has This Been Tested**

N.A.

**Checklist**

N.A.

davidberenstein1957 committed:
docs: initial iteration updated readme

# Conflicts:
#	README.md

davidberenstein1957 committed:
docs: evaluate new structure readme.md

# Conflicts:
#	README.md

davidberenstein1957 committed:
chore: wrapped up cheatsheet for readme

# Conflicts:
#	README.md

davidberenstein1957 committed:
fix: separated code-blocks

davidberenstein1957 committed:
fix: removed iframe and shortened bash

davidberenstein1957 committed:
fix: resolve wrongly rendered video

davidberenstein1957 committed:
fix: added active learning Iframe

davidberenstein1957 committed:
fix: added active learning video

davidberenstein1957 committed:
fix: resolved wrong hyperlinks

# Conflicts:
#	README.md

davidberenstein1957 committed:
fix: resolved typo
chore: added URLs to images

davidberenstein1957 committed:
chore: added a sub-title

# Conflicts:
#	README.md

davidberenstein1957 committed:
fix: resolved some typo's

davidberenstein1957 committed:
chore: added quickstart overview

davidberenstein1957 committed:
docs: added text2text ifo

Commit:
docs: create llms practical guide

Commit:
docs: update practical guide

Commit:
docs: update practical guide

Commit:
docs: create dolly tutorial

Commit:
docs: update practical guide

Commit:
docs: practical guide updates and images

davidberenstein1957 committed:
chore: wrapped up first version of cheatsheet

davidberenstein1957 committed:
chore: removed local and global references connect to argilla

davidberenstein1957 committed:
fix: added correct version entrypoints

davidberenstein1957 committed:
fix: resolved mistune misversioning

Commit:
docs: dolly tutorial first draft

davidberenstein1957 committed:
fix: aligned changed to package with local version

Commit:
docs: guide rename

davidberenstein1957 committed:
fix: update mdit-py-plugins

davidberenstein1957 committed:
fix: changed myst-parser and mistuen

dvsrepo committed:
Create folder for LLM images

dvsrepo committed:
Add sft flow svg (test)

dvsrepo committed:
Preview LLM introduction

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

dvsrepo committed:
Update light/dark compatible diagram

davidberenstein1957 committed:
chore: added re-usable _common tables
chore: create database migration config
chore: added info to tiles
chore: simplified guide names
chore: revamped what is argilla
chore: added separate cheatsheet

Commit:
docs: disagreement section

Commit:
docs: update draft

Commit:
docs: images and table formatting

Commit:
docs: tutorial stats draft

Commit:
docs: comment html table

nataliaElv committed:
Docs/llms-practical-guide (#2955)

# Description

Includes a how-to guide to LLM projects and a notebook tutorial on how
to curate a public dataset like Dolly.

**Type of change**

- [x] Documentation update

**Checklist**

- [ ] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [ ] follows the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

Commit:
docs: correct links and structure in how-to guide

Commit:
docs: disagreement metrics

davidberenstein1957 committed:
chore: added how to guides to docs

Commit:
docs: curation tutorial stats

Commit:
docs: improved formatting

Commit:
docs: eliminate bullet points from cards

Commit:
docs: add feedback task to cheatsheet tabs

Commit:
fix: broken link

Commit:
docs: code refactor

Commit:
fix: callout format

Commit:
docs: dolly tutorial cleaning and add images

alvarobartt commented on docs/_source/guides/llms/practical_guides/collect_responses.ipynb:
At the beginning we should mention that the user is supposed to `rg.init()` if using an Argilla Server different than `localhost`, as we're pulling a `FeedbackDataset` from Argilla. At the same time, we should mentioned that `rg.init` is not mandatory unless we call either `push_to_argilla` and/or `from_argilla`.

alvarobartt commented on docs/_source/guides/llms/practical_guides/collect_responses.ipynb:
I would remove the last section from this Jupyter Notebook, and rename the one `create_and_import_dataset.ipynb` to be `create_import_and_export.ipynb` to add the "Export or publish dataset" there, as this Jupyter Notebook already starts off with a previously published dataset in Argilla, as it's using the `from_argilla` class method.

alvarobartt commented on docs/_source/guides/llms/practical_guides/collect_responses.ipynb:
Additionally, regarding the export, the JSON-export mentioned in this Jupyter Notebook is not supported and will fail, so on, I'd suggest to recommend either `push_to_argilla` or `push_to_huggingface`, and additionally, if the user wants to store it locally, I'd recommend something like:

```python
hfds = ds.format_as("datasets")
hfds.save_to_disk("dataset")
hfds.to_csv("dataset.csv")
hfds.to_parquet("dataset.parquet")
hfds.to_pandas()
```

Which implies converting the `rg.FeedbackDataset` to a HuggingFace Dataset first, and then relying on their pre-implemented functions to export to different formats, to avoid re-inventing the wheel.

alvarobartt commented on docs/_source/guides/llms/practical_guides/define_task.ipynb:
To keep the conversion to `pandas` within HuggingFace Datasets, I would just call `ds.to_pandas()`, since we're already importing the dataset from HuggingFace, and there's a built-in method for that

alvarobartt commented on docs/_source/guides/llms/practical_guides/define_task.ipynb:
I think here we're missing a section to define the fields, so we could create something like "Define fields".

alvarobartt commented on docs/_source/guides/llms/practical_guides/define_task.ipynb:
Also on "Define questions" I think we should link the matching `pydantic.BaseModel` for the rating and the text questions, as well as defining that both the `title` and the `required` params are optional, and those default to capitalized-name, and True, respectively.

alvarobartt replied:
e.g.

* Rating (`rg.RatingQuestion`): ...
* Text (`rg.TextQuestion`): ...

alvarobartt reviewed (dismissed):
Just some minor comments, but I think the `rg.FeedbackDataset` creation could be more clear, as now we need to mix information from two sections to actually be able to create a `rg.FeedbackDataset`.

And also regarding the mentions to import/export, I'd mention first the export and then the import to/from, respectively, either Argilla's or HuggingFace's, since a user cannot import a dataset that hasn't been exported before. And as part of the HuggingFace export, we need to mention that we're also adding a config file named `argilla.cfg` which is required to load the dataset configuration back, the configuration being the fields, questions, and guidelines if applicable.

Also we're missing the mention that at least one field and one question must be `required=True`.

P.S. Sorry for the long review, but I couldn't comment on the files directly since those are Jupyter Notebooks.

Commit:
docs: curate instruction dataset to examples

Commit:
chore: examples page and callout fix

Commit:
fix: curate tutorial bug

alvarobartt committed:
Revision of `rg.FeedbackDataset` documentation (#2970)

# Description

This PR is a revision over the original PR at #2948, with some minor
changes on top of it for readability purposes, as well as adding some
Python client details, that may be hidden for the user, so as to provide
more transparency.

**Type of change**

- [X] Documentation update

---------

Co-authored-by: David Berenstein <david.m.berenstein@gmail.com>

tomaarsen committed:
[Docs] remove unused pages (#2825)

Closes #2566

Hello!

## Pull Request overview
* Remove unused pages (features_all, features_points, missing)
* Fix link to `Setup and Installation` page
* Introduce Unstructured in the list of libraries

## Details
I noticed the following logs when building the docs locally:
```
checking consistency...
C:\code\argilla\docs\_source\_common\features_all.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\_common\features_points.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\_common\missing.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\_common\snippets\training\text-classification\autotrain.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\_common\snippets\training\text-classification\setfit.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\_common\snippets\training\text-classification\spacy.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\_common\snippets\training\text-classification\transformers.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\_common\snippets\training\text2text\autotrain.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\_common\snippets\training\token-classification\autotrain.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\_common\snippets\training\token-classification\spacy.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\_common\snippets\training\token-classification\transformers.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\getting_started\installation\installation.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\tutorials\libraries\unstructured.md: WARNING: document isn't included in any toctree
``` 

### Removing unused pages
This PR removes `features_all.md`, `features_points.md` and
`missing.md`. The last of these three was replaced by
[404.rst](https://github.com/argilla-io/argilla/blob/develop/docs/_source/404.rst).
Note that these files could still be found via searching. For example,
if I search for "Features", then the first result is:

![image](https://user-images.githubusercontent.com/37621491/235898787-18b16a5a-94b3-42d8-bb79-b3065fc56eca.png)

Which is this page:
https://docs.argilla.io/en/latest/_common/features_all.html (i.e.
features_all). As you can tell, all of the links on this page are
broken.

### Installation
```
C:\code\argilla\docs\_source\getting_started\installation\installation.md: WARNING: document isn't included in any toctree
```
Because of this PR, you can again find this page via
https://docs.argilla.io/en/latest/community/developer_docs.html. The
only other way is via searching for it. This does not seem ideal.

@davidberenstein1957 @nataliaElv Should this file be removed, or should
it be placed on the table of contents tree? The current situation (i.e.
almost inaccessible except via searcing) doesn't seem ideal.

### Introduce Unstructured in the list of libraries
```
C:\code\argilla\docs\_source\tutorials\libraries\unstructured.md: WARNING: document isn't included in any toctree
```
Due to this PR, Unstructured will appear in the list here:
https://docs.argilla.io/en/latest/tutorials/libraries/libraries.html as
well as in the TOC tree on the left under "Libraries".

## After this PR
Now, the list is:
```
checking consistency...
C:\code\argilla\docs\_source\_common\snippets\training\text-classification\autotrain.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\_common\snippets\training\text-classification\setfit.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\_common\snippets\training\text-classification\spacy.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\_common\snippets\training\text-classification\transformers.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\_common\snippets\training\text2text\autotrain.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\_common\snippets\training\token-classification\autotrain.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\_common\snippets\training\token-classification\spacy.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\_common\snippets\training\token-classification\transformers.md: WARNING: document isn't included in any toctree
C:\code\argilla\docs\_source\getting_started\installation\installation.md: WARNING: document isn't included in any toctree
```
Which seems better. The snippets shouldn't be included, so that's right,
and the installation page isn't in the TOC yet as mentioned above.

**Type of change**

- [x] Documentation update

**How Has This Been Tested**

N\A

**Checklist**
N\A

- Tom Aarsen

davidberenstein1957 committed:
fix: remove syncing tasks

Commit:
docs: closes Docs: Dolly cleaning tutorial #2928

leiyre committed:
Style: update card and tabs styles for LLM docs (#2972)

# Description

This PR updates styles for LLM docs

- [x] Custom color and size for tabs
- [x] Update card title color, border, shadow and margin

**Type of change**

- [X] Styles

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

davidberenstein1957 committed:
chore: updated colums for hub pages

Commit:
docs: how-to new draft

Commit:
docs: first three pages new version

Commit:
docs: how-to final structure

Commit:
docs: new disagreement plot

nataliaElv committed:
Merge pull request #2974 from argilla-io:docs/llm-guides-reviewed

docs: llm how-to guide review

Commit:
docs: check links and tabs

dvsrepo reviewed (commented):
This is looking good! 

As a quick comment: I feel we should remove the repetitive Next steps block (it's outdated and makes the user feel like going somewhere else, while they should keep reading the next section). @nataliaElv is this something you could tackle or is @davidberenstein1957 ?

Commit:
docs: add references to how-to guides

Commit:
docs: fix typo

Commit:
docs: add missing output

davidberenstein1957 committed:
fix: remove feedbackdataset from managed data

davidberenstein1957 committed:
docs: remove next steps at epilog

dvsrepo committed:
Test svg image fonts

dvsrepo committed:
Adds first version of conceptual guide

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

dvsrepo committed:
Add new svg, still testing

dvsrepo committed:
Merge branch 'docs/llm-document-structure' of github.com:argilla-io/argilla into docs/llm-document-structure

* 'docs/llm-document-structure' of github.com:argilla-io/argilla:
  [pre-commit.ci] auto fixes from pre-commit.com hooks for more information, see https://pre-commit.ci

dvsrepo committed:
Simplify headers for toc

dvsrepo committed:
Keep trying, don't give up

dvsrepo committed:
Cleans up not ready examples, some improvements to conceptual guide

Commit:
docs: title change

Commit:
docs: test tip style

Commit:
docs: tip another test

dvsrepo committed:
Three figures, some more content

Commit:
docs: yet another styling test

Commit:
Merge branch 'docs/llm-document-structure' of github.com:argilla-io/argilla into docs/llm-document-structure

dvsrepo committed:
Add data model figure

dvsrepo committed:
Merge branch 'docs/llm-document-structure' of github.com:argilla-io/argilla into docs/llm-document-structure

* 'docs/llm-document-structure' of github.com:argilla-io/argilla:
  docs: yet another styling test

dvsrepo committed:
Add correct link

dvsrepo committed:
Add end-to-end completion collection workflow

dvsrepo committed:
Add draft end-to-end sft stage

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

dvsrepo committed:
Add more content to conceptual guide

dvsrepo committed:
Merge branch 'docs/llm-document-structure' of github.com:argilla-io/argilla into docs/llm-document-structure
* 'docs/llm-document-structure' of github.com:argilla-io/argilla:
  [pre-commit.ci] auto fixes from pre-commit.com hooks for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

dvsrepo committed:
Use demonstration

dvsrepo committed:
Merge branch 'docs/llm-document-structure' of github.com:argilla-io/argilla into docs/llm-document-structure
* 'docs/llm-document-structure' of github.com:argilla-io/argilla:
  [pre-commit.ci] auto fixes from pre-commit.com hooks for more information, see https://pre-commit.ci

dvsrepo committed:
Refine images

dvsrepo committed:
Restructure conceptual guides

dvsrepo committed:
Fix links and structure

dvsrepo committed:
Polish intro

dvsrepo committed:
Improve index

dvsrepo committed:
Restructure rlhf+sft+rm

Commit:
docs: fix tip styles

dvsrepo committed:
Figure with caption

dvsrepo committed:
Merge branch 'docs/llm-document-structure' of github.com:argilla-io/argilla into docs/llm-document-structure

* 'docs/llm-document-structure' of github.com:argilla-io/argilla:
  docs: fix tip styles

dvsrepo committed:
Figure without caption

dvsrepo committed:
Figure with html

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

davidberenstein1957 committed:
chore: updated dependency for huggingface_hub

Commit:
docs: yet another tip style test

davidberenstein1957 committed:
chore: added docs for peft and autotrain, but without autotrain confi… (#2975)

…g update code

# Description

Add documentation for #2941  and #2940 

**Type of change**


- [X] Documentation update

**How Has This Been Tested**
N.A.

leiyre committed:
style: video height for mobile

davidberenstein1957 committed:
chore: updated to single column readme

davidberenstein1957 committed:
docs: added h3 titles to readme

davidberenstein1957 committed:
docs: removed table format

Commit:
docs: how-to guides to md

Commit:
Merge branch 'docs/llm-document-structure' of github.com:argilla-io/argilla into docs/llm-document-structure

davidberenstein1957 committed:
docs: added LLM support to readme

davidberenstein1957 committed:
docs: fixed broken link to LLM docs

davidberenstein1957 committed:
docs: added hr lines to readme

davidberenstein1957 committed:
docs: added horizontal line above per section

dvsrepo committed:
Finished sft section

dvsrepo committed:
resolve conflict

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

dvsrepo committed:
Add reward modelling section

dvsrepo committed:
Merge branch 'docs/llm-document-structure' of github.com:argilla-io/argilla into docs/llm-document-structure

* 'docs/llm-document-structure' of github.com:argilla-io/argilla:
  [pre-commit.ci] auto fixes from pre-commit.com hooks for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

dvsrepo committed:
Data model and several improvements

dvsrepo committed:
Merge branch 'docs/llm-document-structure' of github.com:argilla-io/argilla into docs/llm-document-structure

* 'docs/llm-document-structure' of github.com:argilla-io/argilla:
  [pre-commit.ci] auto fixes from pre-commit.com hooks for more information, see https://pre-commit.ci

dvsrepo committed:
Fix dm

Commit:
docs: improved dolly tutorial

Commit:
docs: update links

dvsrepo committed:
Review hub page and rlfh guide

dvsrepo committed:
Merge branch 'docs/llm-document-structure' of github.com:argilla-io/argilla into docs/llm-document-structure

* 'docs/llm-document-structure' of github.com:argilla-io/argilla:
  docs: update links
  docs: improved dolly tutorial

dvsrepo committed:
More review

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

davidberenstein1957 committed:
docs: added initial fine-tuning docs

davidberenstein1957 committed:
docs: some formatting updated supervised fine tuning

davidberenstein1957 committed:
docs: formatting correction

dvsrepo committed:
Fix table line breaks

dvsrepo committed:
Merge branch 'docs/llm-document-structure' of github.com:argilla-io/argilla into docs/llm-document-structure
* 'docs/llm-document-structure' of github.com:argilla-io/argilla:
  docs: formatting correction
  docs: some formatting updated supervised fine tuning
  docs: added initial fine-tuning docs
  [pre-commit.ci] auto fixes from pre-commit.com hooks for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

davidberenstein1957 committed:
docs: moved pre-training to end of file

dvsrepo committed:
Review conceptual guides hub

dvsrepo committed:
Merge branch 'docs/llm-document-structure' of github.com:argilla-io/argilla into docs/llm-document-structure

* 'docs/llm-document-structure' of github.com:argilla-io/argilla:
  docs: moved pre-training to end of file
  [pre-commit.ci] auto fixes from pre-commit.com hooks for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

dvsrepo committed:
Final review conceptual guides hub

dvsrepo committed:
Merge branch 'docs/llm-document-structure' of github.com:argilla-io/argilla into docs/llm-document-structure
* 'docs/llm-document-structure' of github.com:argilla-io/argilla:
  [pre-commit.ci] auto fixes from pre-commit.com hooks for more information, see https://pre-commit.ci

alvarobartt commented on tmp_trainer/runs/May30_12-43-52_Davids-MacBook-Pro.local/1685443432.482243/events.out.tfevents.1685443432.Davids-MacBook-Pro.local.25465.1:
Watch out @davidberenstein1957! You uploaded a lot of files under `tmp_trainer/runs` 😩

davidberenstein1957 committed:
refactor: removed training run events

alvarobartt committed:
docs: refactor code-blocks that were looping over `records` as a `dict` instead of `pydantic.BaseModel` (#3036)

# Description

As recently discussed with @frascuchon we should return the
`FeedbackDataset.records` as `pydantic.BaseModel`s instead of Python
dictionaries, since those are more user friendly and we should keep the
validation layer for future releases.

**Type of change**

- [X] Breaking change (fix or feature that would cause existing
functionality to not work as expected)
- [X] Refactor (change restructuring the codebase without changing
functionality)
- [X] Documentation update

**How Has This Been Tested**

- [X] Re-run all the Jupyter Notebooks and scripts to make sure those
work, even fixed some that were not working before (neither with the
previous approach)

**Checklist**

- [X] I have merged the original branch into my forked branch
- [X] I made corresponding changes to the documentation

**Missing**

In `docs/_source/guides/llms/conceptual_guides/rm.md` there's a
code-block that mentions triplets, and that's assuming that some fields
are created in the `FeedbackDataset` while in the definition on a
code-block above those are not created, is that an extra piece of code,
or should we create the fields `response 1`, `response 2`, and `response
3` mentioned there?

---------

Co-authored-by: Francisco Aranda <francis@argilla.io>

davidberenstein1957 committed:
Docs/records are basemodel instead of dict (#3040)

# Description

Resolved some typo's and added doc changes.

---------

Co-authored-by: Alvaro Bartolome <alvaro@argilla.io>
Co-authored-by: Francisco Aranda <francis@argilla.io>

davidberenstein1957 committed:
docs: updated fine-tuning docs

frascuchon committed:
Merge branch 'releases/1.8.0' into docs/llm-document-structure
